sipXyealink/src/org/sipfoundry/sipxconfig/phone/yealink/YealinkPhone.java

```diff
@@ -215,6 +223,18 @@ public boolean getHasHD() {
     public String getDirectedCallPickupString() {
         return getPhoneContext().getPhoneDefaults().getDirectedCallPickupCode();
     }
+    
+    public String getPhonebookType(){
+        //Setting setting = getSettings().getSetting(PHONEBOOKTYPE);
+        Setting settings = getSettings();
+        Setting setting = settings.getSetting(PHONEBOOKTYPE);
+        String settingValue = (null == setting ? null : (setting.getValue() == null ? null : setting
+                        .getValue()));
+        if (!StringUtils.isEmpty(settingValue)) {
+                    return settingValue;
+                }
+        return "2";
```

Comment:
Please use a Constant so code to be more readable.
I can't think of a better time than first time submitting code to run precommit / code style ;)